shared/routing_scope.go

```diff
@@ -0,0 +1,25 @@
+package shared
+
+type RoutingScope struct {
+	rack       string
+	datacenter string
+	fallback   *RoutingScope
```

Comment:
You're introducing this "fallback" but not using it, and the PR is supposed to be just about logging ;-)